http-api/src/main/java/io/avaje/http/api/NimaJsonbStream.java

```diff
@@ -0,0 +1,50 @@
+package io.avaje.http.api;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import io.helidon.nima.webserver.http.ServerResponse;
+
+public class NimaJsonbStream extends OutputStream {
+
+  private final ServerResponse response;
+  private OutputStream outputStream;
+  private byte[] firstBuffer;
+  private int firstOff;
+  private int firstLen;
+  private boolean chunked;
+
+  public NimaJsonbStream(ServerResponse response) {
+    this.response = response;
+  }
+
+  @Override
+  public void write(int b) throws IOException {}
+
+  @Override
+  public void write(byte[] b, int off, int len) throws IOException {
+    if (chunked) {
+      outputStream.write(b, off, len);
+    } else if (firstBuffer == null) {
+      firstBuffer = b;
```

Comment:
I think there is a subtle issue here in that the buffer will get reused / overwritten (if there are 2 or more "buffer writes" / if content exceeds 4k).

That is, I don't think we can keep a reference to b, off, len ... in the form of firstBuffer etc. We'd need to make a defensive copy as we don't know if this was the "first and last write" or "just the first or many writes".